src/cursor/cur_version.c

```diff
@@ -434,17 +437,20 @@ __curversion_next_single_key(WT_CURSOR *cursor)
                     durable_stop_ts = WT_TS_MAX;
                     version_prepare_state = 0;
                 } else
-                    version_prepare_state = cbt->upd_value->tw.prepare;
+                    version_prepare_state = WT_TIME_WINDOW_HAS_PREPARE(&(cbt->upd_value->tw));
             }
 
             WT_ERR(__curversion_set_value_with_format(cursor, WT_CURVERSION_METADATA_FORMAT,
-              cbt->upd_value->tw.start_txn, cbt->upd_value->tw.start_ts,
-              cbt->upd_value->tw.durable_start_ts, stop_txn, stop_ts, durable_stop_ts,
-              WT_UPDATE_STANDARD, version_prepare_state, 0, WT_CURVERSION_DISK_IMAGE));
+              cbt->upd_value->tw.start_txn,
+              WT_MAX(cbt->upd_value->tw.start_ts, cbt->upd_value->tw.start_prepare_ts),
```

Comment:
It should be if we have prepared, we return the prepare ts otherwise it's the start ts